daemon/containerd/image_snapshot.go

```diff
@@ -59,29 +60,44 @@ func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, parentIma
 		parentSnapshot = identity.ChainID(diffIDs).String()
 	}
 
-	// Add a lease so that containerd doesn't garbage collect our snapshot
 	ls := i.client.LeasesService()
 	lease, err := ls.Create(ctx, leases.WithID(id))
 	if err != nil {
 		return err
 	}
-	if err := ls.AddResource(ctx, lease, leases.Resource{
-		ID:   id,
-		Type: "snapshots/" + i.StorageDriver(),
-	}); err != nil {
-		return err
-	}
+	ctx = leases.WithLease(ctx, lease.ID)
 
 	snapshotter := i.client.SnapshotService(i.StorageDriver())
 
+	if err := i.prepareInitLayer(ctx, id, parentSnapshot, setupInit); err != nil {
```

Comment:
`setupInit` is always needed yes

Reply:
The current version doesn't use it though right? My understanding is that is no longer necessary from a runtime perspective and only useful to cleanup this diff case.

Reply:
Ah sorry, I misunderstood your question, so yeah, it would seem that this is not really needed for the runtime. Let me see if I can switch this up and prepare a snapshot with the init layer only when diffing, see where that takes us

Reply:
OK so the init layer is indeed only needed for diffing, I tried not adding it and juggling around with the snapshots but at the end of the day the root in `containerRoot` is missing things (like `/proc` etc).

Reply:
Are those parts that are missing the "default mounts"? Wondering if in that case we could take that list and exclude them 🤔

Reply:
If we exclude them then we won't know if things changed, for example, what would happen if I `rm -fr /etc` in my container? Removing those from the diff would mean the diff is wrong